include/copentime/rationalTime.h

```diff
@@ -4,58 +4,84 @@
 #include <stdbool.h>
 
 #ifdef __cplusplus
-extern "C"
-{
+# define OTIO_API extern "C"
+#else
+# define OTIO_API
 #endif
-    struct RationalTime;
-    typedef struct RationalTime RationalTime;
-
-    typedef enum
-    {
-        OpenTime_IsDropFrameRate_InferFromRate = -1,
-        OpenTime_IsDropFrameRate_ForceNo       = 0,
-        OpenTime_IsDropFrameRate_ForceYes      = 1,
-    } OpenTime_IsDropFrameRate_;
-    typedef int   OpenTime_IsDropFrameRate;
-    RationalTime* RationalTime_create(double value, double rate);
-    bool         RationalTime_is_invalid_time(RationalTime* self);
-    double        RationalTime_value(RationalTime* self);
-    double        RationalTime_rate(RationalTime* self);
-    RationalTime* RationalTime_rescaled_to(RationalTime* self, double new_rate);
-    RationalTime* RationalTime_rescaled_to_rational_time(
-        RationalTime* self, RationalTime* rt);
-    double
-           RationalTime_value_rescaled_to_rate(RationalTime* self, double new_rate);
-    double RationalTime_value_rescaled_to_rational_time(
-        RationalTime* self, RationalTime* rt);
-    bool RationalTime_almost_equal(
-        RationalTime* self, RationalTime* other, double delta);
-    RationalTime* RationalTime_duration_from_start_end_time(
-        RationalTime* start_time, RationalTime* end_time_exclusive);
-    bool         RationalTime_is_valid_timecode_rate(double rate);
-    RationalTime* RationalTime_from_frames(double frame, double rate);
-    RationalTime* RationalTime_from_seconds(double seconds);
-    RationalTime* RationalTime_from_timecode(
-        const char* timecode, double rate, OpenTimeErrorStatus* error_status);
-    RationalTime* RationalTime_from_time_string(
-        const char* time_string, double rate, OpenTimeErrorStatus* error_status);
-    int    RationalTime_to_frames(RationalTime* self);
-    int    RationalTime_to_frames_with_rate(RationalTime* self, double rate);
-    double RationalTime_to_seconds(RationalTime* self);
-    const char* RationalTime_to_timecode(
-        RationalTime*        self,
-        double               rate,
+
+struct RationalTime {
+    double value;
+    double rate;
+};
+typedef struct RationalTime RationalTime;
```

Comment:
This is the pattern we've been using but I think we ought to switch to the more compact form. 
```suggestion
typedef struct {
    double value;
    double rate;
} RationalTime;
```

Reply:
How do you forward declare RationalTime ? Might be showing my C ignorance here.